### PR TITLE
faster ci + connector widget improvements + more testing

### PR DESCRIPTION
**Pull request by edublancas**
## Describe your changes

- Refactored CI (more reliable, faster)
- ConnectorWidget loads dsn_filename from JupySQL's configuration
- ConnectorWidget loads stored connnections each time it's initialized
- Adds ConnectorWidget Python logic unit tests
- General improcementes to ConnectorWidget
- Adds missing tests and improves existing ones

## Issue number

Closes #X

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

### CODE REVIEW

File: ui-tests/README.md [hunk 1]
```diff
@@ -48,6 +51,12 @@ will be opened in your browser at the end of the tests execution; see
 [Playwright documentation](https://playwright.dev/docs/test-reporters#html-reporter)
 for configuring that behavior.
 
+### Noes for writing UI tests
+
+- If the UI tests fail, check the recording. If the video shows that no notebook is opened, it might  be that the tests are trying to open a notebook that doesn't exist, this might happen with notebooks in hidden folders
```
Comment (yafimvo) [inline ui-tests/README.md:56]: might  be -> might be (double space)

**Review by yafimvo — CHANGES_REQUESTED** [review]
@edublancas lg. Added some minor comments

Few questions and notes
1. Why did you delete the `environment.yml` file? 
2. I ran `nox --session test` locally; some tests didn't pass. Should it pass?
3. I tried to create a `connections.ini` file in a different path. First I thought I should only specify the path and it will create `connection.ini` file. However, it seems like we can create any file with any extension. Is this the expected behavior?

**Comment by edublancas** [comment]
> @edublancas lg. Added some minor comments
> 
> Few questions and notes
> 
>     1. Why did you delete the `environment.yml` file?

i switched for a requirements.txt because `pip` is faster than `conda`, this speeds up the CI


>     2. I ran `nox --session test` locally; some tests didn't pass. Should it pass?

what error are you getting?


>     3. I tried to create a `connections.ini` file in a different path. First I thought I should only specify the path and it will create `connection.ini` file. However, it seems like we can create any file with any extension. Is this the expected behavior?

it is. you should specify the full path to the file

**Comment by edublancas** [comment]
> Got a bit stuck while testing. Do we need to setup any file?
> Screenshot 2023-08-21 at 8 05 58 PM
> 
> ```
>   479         saved configuration settings
>     480     """
> --> 481     data = load_toml(file_path)
>     482     while section_names:
>     483         section_to_find, sections_from_user = section_names.pop(0), data.keys()
> 
> File ~/miniforge3/envs/jupysql-plugin/lib/python3.11/site-packages/ploomber_core/dependencies.py:40, in requires.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
>      32 @wraps(f)
>      33 def wrapper(*args, **kwargs):
>      34     check_installed(
>      35         pkgs=pkgs,
>      36         name=name or f.__name__,
>      37         extra_msg=extra_msg,
>      38         pip_names=pip_names,
>      39     )
> ---> 40     return f(*args, **kwargs)
> 
> File ~/miniforge3/envs/jupysql-plugin/lib/python3.11/site-packages/sql/util.py:442, in load_toml(file_path)
>     440         content = file.read()
>     441         return toml.loads(content)
> --> 442 except toml.TomlDecodeError as e:
>     443     raise parse_toml_error(e, file_path)
> 
> AttributeError: 'NoneType' object has no attribute 'TomlDecodeError'
> ```

if you look at the traceback, this is coming from jupysql, not jupysql-plugin. my guess is that your environment has an old jupysql version, upgrade and see if it's fixed. if you don't have `toml` installed in the environment, it should suggest you to do so

File: jupysql_plugin/widgets/connections.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    config.read(get_path_to_config_file())
+    return config
+
+
+def _store_connection_details(connection_name, fields):
+    """
+    Stores connection in the config file
+    """
+    # add section test
```
[the remark this answers is not in the document]
Comment (edublancas) [reply jupysql_plugin/widgets/connections.py]: I can't remember. probably some note I forgot to delete while refactoring. deleted: https://github.com/ploomber/jupysql-plugin/pull/63/commits/1dd0bcf5207180cb20934154cf65c1cd88add1ed

File: src/widgets/connector.ts [hunk 3]
```diff
@@ -93,15 +94,15 @@ export class ConnectorView extends DOMWidgetView {
 
                 <div class="connections-guidelines block">
                     <i>
-                        * Connections are loaded from connections.ini file
+                        * Connections are loaded from your connections file (set it with <code>%config SqlMagic.dsn_filename = "path/to/file", default is "~/.jupysql/connections.ini"</code>).
                     </i>
 
                     <i class="no-config-file" style = "display: none;">
-                        * No connections.ini file found. You may need to restart the kernel.
+                        * No conections file found. You may need to restart the kernel.
```
[the remark this answers is not in the document]
Comment (edublancas) [reply src/widgets/connector.ts]: https://github.com/ploomber/jupysql-plugin/pull/63/commits/28189029d68cb120640618c2c3edc808612a8520

File: src/widgets/connector.ts [hunk 4]
```diff
@@ -93,15 +94,15 @@ export class ConnectorView extends DOMWidgetView {
...
                     </i>
 
                     <i class="restart-kernel" style = "display: none;">
-                        * No connections found in connections.ini file. You may need to restart the kernel.
+                        * No connections found in conections file. You may need to restart the kernel.
```
[the remark this answers is not in the document]
Comment (edublancas) [reply src/widgets/connector.ts]: https://github.com/ploomber/jupysql-plugin/pull/63/commits/6faf9f173c27a70ee75eb68c2caf58ab2b1aee2d

File: ui-tests/README.md (hunk 1 above)
Comment (edublancas) [reply ui-tests/README.md]: https://github.com/ploomber/jupysql-plugin/commit/6faf9f173c27a70ee75eb68c2caf58ab2b1aee2d

File: README.md [hunk 5]
```diff
@@ -10,14 +10,17 @@ pip install jupysql-plugin
 ### Development install
 
 ```sh
-conda env create -f environment.yml --force
+ conda create --name jupysql-plugin python=3.11 --channel conda-forge --yes
```
Comment (yafimvo) [inline README.md:13]: indentation

**Review by yafimvo — APPROVED** [review]
lg.


> i switched for a requirements.txt because pip is faster than conda, this speeds up the CI
Got it. I saw you updated the readme file - added a small comment on the indentation.

> what error are you getting?

I ran the tests again and it seems like it's a screenshot comparison issue (tested on Mac) which I guess is ok

[ui-tests.zip](https://github.com/ploomber/jupysql-plugin/files/12409521/ui-tests.zip)

**Comment by brichet** [comment]
@edublancas shouldn't this PR be backported on branch `jupyterlab4` ?

It seems that the following statement is broken with `jupysql>=0.10`, since https://github.com/ploomber/jupysql/pull/803.

https://github.com/ploomber/jupysql-plugin/blob/1e058e7f93d3085b7a7c0eab4f5c4864000b44eb/jupysql_plugin/widgets/connector_widget.py#L7

**Pull request merged**